Exercise_6/Exercise_6.ipynb

```diff
@@ -180,33 +180,69 @@
+    }
+   },
+   "cell_type": "markdown",
+   "id": "278234da",
+   "metadata": {},
+   "source": [
+    "Please open a terminal and go to the directory you get executing the cell above.\n",
+    "\n",
+    "![Open_Terminal.png](attachment:Open_Terminal.png)\n"
    ]
   },
   {
+   "cell_type": "markdown",
+   "id": "a5c93d5d",
+   "metadata": {},
+   "source": [
+    "### Create user token for https\n",
+    "\n",
+    "GitHub only allows authorized users to push to repositories.\n",
+    "Therefore we need to create a GitHub user token.\n",
+    "\n",
+    "Please follow the [description](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-token) provided by GitHub.\n",
+    "\n",
+    "Once you have a token, you can enter it instead of your password when performing Git operations over HTTPS.\n",
+    "\n",
+    "In our case we use the token to perform the `git push` command in the terminal you just opened:\n",
+    "\n",
+    "```\n",
+    "git push origin <branch_name>\n",
```

Comment:
I get a popup window when I trying doing this, telling me "git-credential-osxkeychain wants to use your confidiential information stored in “[github.com](http://github.com/)" in your keychain.".
When I press deny, I can proceed with my username and tokens but I guess we should add a comment about this.

Reply:
Yes we need to test this mechanism very well.

I think it makes sense to focus on what happens on the IAC Jupyter Hub, otherwise we are lost with Mac, Windows, Linux and all the different git version around.